File: week1-cpp/Getting_started_with_images/Splitting_and_merging_channels.cpp
#include <iostream>
#include <opencv2/opencv.hpp>
#include <opencv2/core.hpp>
#include <opencv2/highgui.hpp>
#include "dataPath.hpp"

using namespace std;
using namespace cv;

int main(void)
{
        string imagePath = DATA_PATH + "/images/musk.jpg";
	// Read the image
        Mat img = imread(imagePath);
	// Show the channels
	Mat imgChannels[3];
	split(img, imgChannels);

	// Write the channels
	imwrite("../results/imgBlue.png",imgChannels[0]);
	imwrite("../results/imgGreen.png",imgChannels[1]);
	imwrite("../results/imgRed.png",imgChannels[2]);

	//imshow("Blue Channel",imgChannels[0]);
	//imshow("Green Channel",imgChannels[1]);
	//imshow("Red Channel",imgChannels[2]);
	//waitKey(0);

	return 0;
}

File: week1-python/Getting_started_with_images/Splitting_and_Merging_Channels.py
# Import libraries
import cv2
import numpy as np
from dataPath import DATA_PATH

# Path of the image to be loaded
# Here we are supplying a relative path
imagePath = DATA_PATH + "/images/musk.jpg"

# Read the image
img = cv2.imread(imagePath)

# Split the image into the B,G,R components
b,g,r = cv2.split(img)

# Show the channels
#cv2.imshow("Blue Channel",b)
#cv2.imshow("Green Channel",g)
#cv2.imshow("Red Channel",r)
cv2.imwrite("../results/blueChannel.png",b)
cv2.imwrite("../results/greenChannel.png",g)
cv2.imwrite("../results/redChannel.png",r)

# Merge the individual channels into a BGR image
mergedOutput = cv2.merge((b,g,r))
# Show the merged output
#cv2.imshow("Merged Output",mergedOutput)
#cv2.waitKey(0)
#cv2.destroyAllWindows()
cv2.imwrite("../results/mergedOutput.png",mergedOutput)
